src/app/app.component.ts

```diff
@@ -599,69 +638,81 @@ export class AppComponent implements OnInit, AfterViewInit, CanvasTableSelectLis
     const snackBarRef = this.snackBar.open('Toggling read status...');
     const messageIds = this.canvastable.rows.selectedMessageIds();
 
-
-    const args = {
-      flag: {
-        name: 'seen_flag',
-        value: status ? 1 : 0,
-      },
-      ids: messageIds
-    };
-
-    this.rmm.email.update(args).subscribe(() => {
-        this.searchService.updateIndexWithNewChanges();
-        if (!this.searchService.localSearchActivated) {
-          messageIds.forEach( (id) => {
-               this.rmmapi.messageFlagChangeSubject.next(
-                   new MessageFlagChange(id, status, null)
-               );
-          } );
-        }
+    this.messageActionsHandler.updateMessages(
+      messageIds,
+      // local index / messagelist
+      (msgIds: number[]) => {
+        msgIds.forEach( (id) => {
+          // Updates both index + messagelist
+          this.rmmapi.messageFlagChangeSubject.next(
+            new MessageFlagChange(id, status, null)
+          );
+        });
         this.clearSelection();
-        this.showSelectMarkOpMenu = false;
-        snackBarRef.dismiss();
-    });
+      },
+      // remote api
```

Comment:
Comments similar to these pop up in most of these `updateMessages()` calls – they are helpful, which makes me wonder if we want to enforce this in an API using quasi-named arguments, like so?

```
this.messageActionsHandler.updateMessages(messageIds, {
    local: fun1(),
    remote: fun2()
});
```

What do you think?

Reply:
Given our usage patterns here, it could also have a `finally` optional callback, that'd prevent us from subscribing/piping the result to react to remote changes.